### DISCUSSION TITLE
How to hide credentials from default.json using CI/CD env variables

### DISCUSSION THREAD

**Discussion by larasika** [renovatebot/renovate#12259] [Request Help]
Hello !

Currently in our project (which acts as the "host" of our renovate runner) is using a `default.json` file for configuring renovate on the target repositories.
One of the configuration options that we use is `regexManager` which uses `registryUrlTemplate`. And here as a value, we put a Gitlab Personal Token in the URL.
We want to hide the personal token and store it as a CI/CD environment instead – but how can we inject the variable value to our `default.json` ??

I've tried using `sed` command that is run in the script within `.gitlab-ci.yml` before running the `renovate` command. The idea here is, before the script runs, `default.json` only has a reference/key word called "REGISTRY_URL" (for example) as the value of `registryUrlTemplate` , and then once the script starts running, `sed` command will find this key word and change it to a variable value that is passed in the `.gitlab-ci.yml`.
This works in terms of changing the `default.json`, but when the `renovate` script runs, somehow it will still use the `default.json` that has the key word "REGISTRY_URL", instead of the one that has been changed to the actual variable value.

Could someone help or advise me how we can pass the variable value to `default.json` that will be noticed by renovate ?

Thank you in advance !

**Comment by rarkins** [comment]
Assuming you're talking about using Renovate presets, they are fetched using APIs so any changes you make to it in a pipeline have no effect because they are not fetched from the file system where your modified files presumably lie.

Putting tokens into registry URLs is not a common practice. Usually we recommend `hostRules`.

Can you back up a step and more clearly describe how you are running Renovate and what you're trying to achieve?

For example, are you using our GitLab runner?

What type of datasource needs the credentials?
